### PR TITLE
Update drop_dir to API v2.

### LINKED ISSUE #83: drop_dir
Relevant URLs
```
metadata_url <- "https://api.dropbox.com/1/metadata/auto/"
```

Notes from @scheidec

- Code: https://github.com/karthik/rdrop2/blob/master/R/drop_dir.R
- Change from v1: /1/metadata to v2: /2/files/get_metadata
- Parameters, return values, and metadata have changed
- V1 docs: https://www.dropbox.com/developers-v1/core/docs#metadata
- V2 docs: https://www.dropbox.com/developers/documentation/http/documentation#files-get_metadata
- Metadata mapping: https://www.dropbox.com/developers/reference/migration-guide#metadata-mapping
- bytes -> size
- modified -> server_modified
- mime_type and root are no longer available in v2
- is_dir -> check if .tag=“folder”

### PR DESCRIPTION
**Pull request by ClaytonJY**
Closes #83.

### CODE REVIEW

**Comment by ClaytonJY** [comment]
Only the first piece thus far, plus two additional fixes-on-the side:
- `dtoken` in `drop_upload`
- move old test file to the end of the numbers so it runs last

**Comment by ClaytonJY** [comment]
Turns out I didn't actually need to implement `drop_get_metadata` at all, but I finished it before realizing.

drop_dir now returns things with different names and in a different order than v1, so many tests which used drop_dir needed small edits.

**Comment by ClaytonJY** [comment]
Last commit was very innocent change that managed to break build; I think these are either race conditions (two builds at once making/looking for same folders), or consequences of errors preventing proper cleanup. Maybe I should update those old tests in this PR to use `uuid::UUIDgenerate` like we do elsewhere?

**Comment by ClaytonJY** [comment]
Cheated and re-ran instead to get a pass.

Will make separate issue to revisit old tests post-0.8.

**Comment by ClaytonJY** [comment]
Made a bunch of fixes, one pass one fail, I think due to the race condition mentioned in #112.

File: R/drop_dir.R [hunk 1]
```diff
@@ -1,102 +1,126 @@
...
-#' @param  include_deleted Only applicable when list is set. If this parameter is set to true, then contents will include the metadata of deleted children. Note that the target of the metadata call is always returned even when it has been deleted (with is_deleted set to true) regardless of this flag.
-#' @param  rev If you include a particular revision number, then only the metadata for that revision will be returned.
-#' @template locale
-#' @param include_media_info   If true, each file will include a photo_info dictionary for photos and a video_info dictionary for videos with additional media info. If the data isn't available yet, the string pending will be returned instead of a dictionary.
-#' @template include_membership
-#' @template verbose
+#' @param path path to folder in Dropbox to list contents of. Defaults to the root directory.
+#' @param recursive If true, the list folder operation will be applied recursively to all subfolders and the response will contain contents of all subfolders. Defaults to FALSE.
+#' @param include_media_info If true, FileMetadata.media_info is set for photo and video. Defaults to FALSE.
```
Comment (karthik) [inline R/drop_dir.R]: For this and the next few lines of doc, do you want to use:

If `TRUE`

Just seems like a consistent thing to do as far as making sure even a cursory read should make it clear that it's a logical.  Not a mandatory change.

File: R/drop_dir.R [hunk 2]
```diff
@@ -1,102 +1,126 @@
...
+  httr::stop_for_status(req)
+
+  httr::content(req)
+}
+
+
+#' Helper function for fetching additional results
+#'
+#' @noRd
+drop_list_folder_continue <- function(cursor, dtoken = get_dropbox_token()) {
```
Comment (karthik) [inline R/drop_dir.R]: For `drop_list_entries` and `drop_list_folder_continue` add the @internal tag to denote internal function. Using the internal keyword removes all functions in the associated .Rd file from the documentation index and disables some of their automated tests.

File: R/drop_upload.R [hunk 3]
```diff
@@ -52,7 +52,7 @@ drop_upload <- function(file,
 
   req <- httr::POST(
     url = put_url,
-    httr::config(token = get_dropbox_token()),
+    httr::config(token = dtoken),
```
Comment (karthik) [inline R/drop_upload.R]: 🙏

File: R/drop_utils.R [hunk 4]
```diff
@@ -26,7 +26,7 @@ strip_slashes <- function(path) {
 #' A small function to add a prefix slash
 #' @noRd
 add_slashes <- function(path) {
-  if (length(path) && !grepl("^/", "path")) {
+  if (length(path) && !grepl("^/", path)) {
```
Comment (karthik) [inline R/drop_utils.R]: 👍

File: R/drop_dir.R (hunk 2 above)
Comment (ClaytonJY) [reply R/drop_dir.R]: done!

File: R/drop_dir.R (hunk 1 above)
Comment (ClaytonJY) [reply R/drop_dir.R]: done!

**Comment by karthik** [comment]
All good!

**Pull request merged**